src/NUnitFramework/framework/Constraints/DictionaryContainsKeyConstraint.cs

```diff
@@ -66,31 +73,250 @@ public override string Description
+        }
+
+        /// <summary>
+        /// Flag the constraint to use the supplied boolean-returning delegate.
+        /// </summary>
+        /// <param name="comparer">The supplied boolean-returning delegate to use.</param>
+        [Obsolete(ObsoleteMessage)]
+        public new CollectionItemsEqualConstraint Using<T>(Func<T, T, bool> comparer)
+        {
+            _isDeprecatedMode = true;
+            return base.Using(comparer);
+        }
+
+        #endregion
+
+        private static MethodInfo GetContainsKeyMethod(object keyedItemContainer)
+        {
+            if (keyedItemContainer == null) throw new ArgumentNullException(nameof(keyedItemContainer));
+            var instanceType = keyedItemContainer.GetType();
+
+            var method = FindContainsKeyMethod(instanceType)
+                         ?? instanceType
+                            .GetInterfaces()
+                            .Concat(GetBaseTypes(instanceType))
+                            .Select(FindContainsKeyMethod)
+                            .FirstOrDefault(m => m != null);
+
+            return method;
+        }
+
+        private static MethodInfo FindContainsKeyMethod(Type type)
+        {
+            var methods = type.GetMethods(BindingFlags.Instance | BindingFlags.Public);
+            var method = methods.FirstOrDefault(m =>
+                m.ReturnType == typeof(bool)
+                && m.Name == "ContainsKey"
+                && !m.IsGenericMethod
+                && m.GetParameters().Length == 1);
+
+            if (method == null && type.GetTypeInfo().IsGenericType)
+            {
+#if !NETSTANDARD1_4
+                var definition = type.GetGenericTypeDefinition();
+                var tKeyGenericArg = definition.GetGenericArguments().FirstOrDefault(typeArg => typeArg.Name == "TKey");
+
+                if (tKeyGenericArg != null)
+                {
+                    method = definition
+                             .GetMethods(BindingFlags.Instance | BindingFlags.Public)
+                             .FirstOrDefault(m =>
+                                 m.ReturnType == typeof(bool)
+                                 && m.Name == "Contains"
+                                 && !m.IsGenericMethod
+                                 && m.GetParameters().Length == 1 
+                                 && m.GetParameters()[0].ParameterType == tKeyGenericArg);
+
+                    if (method != null)
+                    {
+                        method = methods.Single(m => m.MetadataToken == method.MetadataToken);
+                    }
+                }
+#else
+                    method = methods.FirstOrDefault(m => 
+                        m.ReturnType == typeof(bool)
+                        && m.Name == "Contains"
+                        && m.IsGenericMethod
+                        && m.GetParameters().Length == 1
+                        && m.GetParameters()[0].ParameterType == type.GetGenericTypeDefinition());
```

Comment:
Great! Indentation is off and `ParameterType == type.GetGenericTypeDefinition()` isn't what we want (parameter type could be `object` if `type.GetGenericTypeDefinition()` is `Dictionary<,>`. Probably don't need a check?